### ISSUE TITLE
kotest fails to start with object instance of AbstractProjectConfig

### ISSUE THREAD

**Issue by cloudshiftchris** [kotest/kotest#3676]
kotest 5.7.1

When running `kotest` with configuration auto-scan disabled and a configuration object (not class) fqn specified in `kotest.properties` it fails to start, complaining (cryptically) about being unable to instantiate the class.

```
kotest.framework.dump.config=true
kotest.framework.classpath.scanning.config.disable=true
kotest.framework.classpath.scanning.autoscan.disable=true
kotest.framework.config.fqn=***.testing.KotestConfig
```

This is due to the configuration loading [here](https://github.com/kotest/kotest/blob/402597ca4bf581f10df2f3d062a2427e0de2d005/kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/config/loadProjectConfigFromClassname.kt#L12) using `newInstance`.

Changing to `class` instead of `object` for the configuration resolves the issue.

The [documentation](https://kotest.io/docs/framework/project-config.html) states:

> Project level configuration can be used by creating an object or class that extends from AbstractProjectConfig.

... and then [further](https://kotest.io/docs/framework/project-config.html#runtime-detection) states:

> Once auto scanning is disabled, if you wish to still use project config, you can specify a well known class name which Kotest will reflectively instantiate. 

Which contradicts the earlier statement.

It seems like the fqn loading could be smart enough to detect if it's an `object` or `class` and behave accordingly, with a single unambiguous statement on extending `AbstractProjectConfig`.

**Comment by cloudshiftchris** [comment]
imo, find `object` configuration conceptually cleaner: no guesswork at who (magically) instantiates it, when it is instantiated, are there multiple instances instantiated, and so on.

Perhaps its worth deprecating `class` configuration for next major release.

**Comment by sksamuel** [comment]
It should support both classes and objects. If you're experiencing otherwise, that's a bug.

**Comment by cloudshiftchris** [comment]
This is the cryptic exception when trying to instantiate an `object` configuration instance using fqn property:
```
Exception in thread "main" java.lang.IllegalAccessException: class foo.FileKt cannot access a member of class foo.MyConfig with modifiers "private"
 at jdk.internal.reflect.Reflection.newIllegalAccessException (:-1) 
 at java.lang.reflect.AccessibleObject.checkAccess (:-1) 
 at java.lang.reflect.Constructor.newInstanceWithCaller (:-1) 
```

It looks like `loadProjectConfigFromClassnameJVM` was designed to only accept `class` types.

[Repro](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS45LjEwIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoicGFja2FnZSBmb29cblxuZnVuIG1haW4oKSB7XG4gICAgbG9hZFByb2plY3RDb25maWdGcm9tQ2xhc3NuYW1lSlZNKClcbn1cblxucHVibGljIG9iamVjdCBNeUNvbmZpZyB7XG4gICAgXG59XG5cbmZ1biBsb2FkUHJvamVjdENvbmZpZ0Zyb21DbGFzc25hbWVKVk0oKSB7XG4gICBDbGFzcy5mb3JOYW1lKFwiZm9vLk15Q29uZmlnXCIpLm5ld0luc3RhbmNlKCkgXG59XG5cbi8qXG5mdW4gbG9hZFByb2plY3RDb25maWdGcm9tQ2xhc3NuYW1lSlZNKCk6IEFic3RyYWN0UHJvamVjdENvbmZpZz8ge1xuICAgdmFsIGZxbiA9IFN5c3RlbS5nZXRQcm9wZXJ0eShLb3Rlc3RFbmdpbmVQcm9wZXJ0aWVzLmNvbmZpZ3VyYXRpb25DbGFzc05hbWUpID86IHJldHVybiBudWxsXG4gICByZXR1cm4gQ2xhc3MuZm9yTmFtZShmcW4pLm5ld0luc3RhbmNlKCkgYXMgQWJzdHJhY3RQcm9qZWN0Q29uZmlnXG59XG4qLyJ9) in Kotlin playground.

**Comment by cloudshiftchris** [comment]
One way to check if an `object` (depends on Kotlin-reflect):

```
private fun loadProjectConfigFromClassnameJVM() {
    val klazz = Class.forName("***.testing.MyConfig").kotlin
    println(klazz)
    println(klazz.objectInstance)

    val instance = when {
        klazz.objectInstance != null -> klazz.objectInstance
        else -> klazz.newInstanceNoArgConstructor()
    }
    println(instance)
}
```

Edit:

It looks like `JvmReflection.newInstanceNoArgConstructorOrObjectInstance` already exists for this case.
And there is `instantiateOrObject` that is used to load the configuration class/object from scanning, without using reflection (method confusingly names its param `kclass` though it is `Class<T>`)

**Comment by sksamuel** [comment]
This was fixed in 5.8.0
